var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -97,30 +100,38 @@ class Geant4(CMakePackage):
     depends_on("python@3:", when="+python")
     extends("python", when="+python")
 
-    for std in _cxxstd_values:
-        # CLHEP version requirements to be reviewed
-        depends_on("clhep@2.4.6.0: cxxstd=" + std, when="@11.1: cxxstd=" + std)
-
-        depends_on("clhep@2.4.5.1: cxxstd=" + std, when="@11.0.0: cxxstd=" + std)
-
-        depends_on("clhep@2.4.4.0: cxxstd=" + std, when="@10.7.0: cxxstd=" + std)
-
-        depends_on("clhep@2.3.3.0: cxxstd=" + std, when="@10.3.3:10.6 cxxstd=" + std)
-
-        # Spack only supports Xerces-c 3 and above, so no version req
-        depends_on("xerces-c netaccessor=curl cxxstd=" + std, when="cxxstd=" + std)
-
-        # Vecgeom specific versions for each Geant4 version
-        depends_on("vecgeom@1.2.0: cxxstd=" + std, when="@11.1: +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@1.1.18:1.1 cxxstd=" + std, when="@11.0.0:11.0 +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@1.1.8:1.1 cxxstd=" + std, when="@10.7.0:10.7 +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@1.1.5 cxxstd=" + std, when="@10.6.0:10.6 +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@1.1.0 cxxstd=" + std, when="@10.5.0:10.5 +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@0.5.2 cxxstd=" + std, when="@10.4.0:10.4 +vecgeom cxxstd=" + std)
-        depends_on("vecgeom@0.3rc cxxstd=" + std, when="@10.3.0:10.3 +vecgeom cxxstd=" + std)
-
-        # Boost.python, conflict handled earlier
-        depends_on("boost@1.70: +python cxxstd=" + std, when="+python cxxstd=" + std)
+    # CLHEP version requirements to be reviewed
+    depends_on("clhep@2.4.6.0:", when="@11.1:")
+    depends_on("clhep@2.4.5.1:", when="@11.0.0:")
+    depends_on("clhep@2.4.4.0:", when="@10.7.0:")
+    depends_on("clhep@2.3.3.0:", when="@10.3.3:10.6")
+
+    # Vecgeom specific versions for each Geant4 version
+    with when("+vecgeom"):
+        depends_on("vecgeom@1.2.0:", when="@11.1:")
+        depends_on("vecgeom@1.1.18:1.1", when="@11.0.0:11.0")
+        depends_on("vecgeom@1.1.8:1.1", when="@10.7.0:10.7")
+        depends_on("vecgeom@1.1.5", when="@10.6.0:10.6")
+        depends_on("vecgeom@1.1.0", when="@10.5.0:10.5")
+        depends_on("vecgeom@0.5.2", when="@10.4.0:10.4")
+        depends_on("vecgeom@0.3rc", when="@10.3.0:10.3")
+
+    for _cxxstd in _cxxstd_values:
+        for _v in (
+            _cxxstd
+            if isinstance(_cxxstd, _ConditionalVariantValues)
+            else [Value(_cxxstd, when="")]
+        ):
+            (std, when) = (_v.value, _v.when)
+
+            depends_on(f"clhep cxxstd={std}", when=f"{when} cxxstd={std}")
```

Comment:
I think `f""` might be too new for some of the versions of python spack must support?

Reply:
These are fine now since f strings were added in python 3.7, which is the spack minimum.